src/TextField/TextFieldLabel.jsx

```diff
@@ -95,6 +92,5 @@ const TextFieldLabel = (props) => {
 };
 
 TextFieldLabel.propTypes = propTypes;
-TextFieldLabel.defaultProps = defaultProps;
 
-export default TextFieldLabel;
+export default shouldUpdate(withShallow('style'))(TextFieldLabel);
```

Comment:
I'm unsure about the benefit of looping through the style keys to prevent a wasted render.
I think that having some style memoization would be more efficient.